packages/three-vrm-springbone/src/VRMSpringBoneJoint.ts

```diff
@@ -241,64 +232,42 @@ export class VRMSpringBoneJoint {
-    let matrixWorldToCenter = this._getMatrixWorldToCenter(_matA);
-    _centerSpacePosition.copy(_worldSpacePosition).applyMatrix4(matrixWorldToCenter);
-    const quatWorldToCenter = _quatA.setFromRotationMatrix(matrixWorldToCenter);
-
-    // Get parent matrix in center space
-    const centerSpaceParentMatrix = _matB.copy(matrixWorldToCenter).multiply(this._parentMatrixWorld);
-
-    // Get boneAxis in center space
-    const centerSpaceBoneAxis = _v3B
+    // Get boneAxis in world space
+    const worldSpaceBoneAxis = _v3B
       .copy(this._boneAxis)
-      .applyMatrix4(this._initialLocalMatrix)
-      .applyMatrix4(centerSpaceParentMatrix)
-      .sub(_centerSpacePosition)
-      .normalize();
-
-    // gravity in center space
-    const centerSpaceGravity = _v3C.copy(this.settings.gravityDir).applyQuaternion(quatWorldToCenter).normalize();
-
-    const matrixCenterToWorld = this._getMatrixCenterToWorld(_matA);
+      .transformDirection(this._initialLocalMatrix)
+      .transformDirection(this._parentMatrixWorld);
 
     // verlet積分で次の位置を計算
     _nextTail
+      // Determine inertia in center space
       .copy(this._currentTail)
-      .add(
-        _v3A
-          .copy(this._currentTail)
-          .sub(this._prevTail)
-          .multiplyScalar(1 - this.settings.dragForce),
-      ) // 前フレームの移動を継続する(減衰もあるよ)
-      .add(_v3A.copy(centerSpaceBoneAxis).multiplyScalar(this.settings.stiffness * delta)) // 親の回転による子ボーンの移動目標
-      .add(_v3A.copy(centerSpaceGravity).multiplyScalar(this.settings.gravityPower * delta)) // 外力による移動量
-      .applyMatrix4(matrixCenterToWorld); // tailをworld spaceに戻す
+      .add(_v3A.subVectors(this._currentTail, this._prevTail).multiplyScalar(1 - this.settings.dragForce)) // 前フレームの移動を継続する(減衰もあるよ)
+      // Convert center space to world space
+      .applyMatrix4(this._getMatrixCenterToWorld()) // tailをworld spaceに戻す
+      // Apply stiffness and gravity in world space
+      .addScaledVector(worldSpaceBoneAxis, this.settings.stiffness * delta) // 親の回転による子ボーンの移動目標
+      .addScaledVector(this.settings.gravityDir, this.settings.gravityPower * delta); // 外力による移動量
 
     // normalize bone length
+    _worldSpacePosition.setFromMatrixPosition(this.bone.matrixWorld);
     _nextTail.sub(_worldSpacePosition).normalize().multiplyScalar(this._worldSpaceBoneLength).add(_worldSpacePosition);
 
     // Collisionで移動
     this._collision(_nextTail);
 
     // update prevTail and currentTail
-    matrixWorldToCenter = this._getMatrixWorldToCenter(_matA);
-
     this._prevTail.copy(this._currentTail);
-    this._currentTail.copy(_v3A.copy(_nextTail).applyMatrix4(matrixWorldToCenter));
+    this._currentTail.copy(_nextTail).applyMatrix4(this._getMatrixWorldToCenter());
 
     // Apply rotation, convert vector3 thing into actual quaternion
     // Original UniVRM is doing center unit calculus at here but we're gonna do this on local unit
-    const worldSpaceInitialMatrixInv = mat4InvertCompat(
```

Comment:
I don't know why we still have `mat4InvertCompat` here... Thank you for deleting this.